Packs/QRadar/Integrations/QRadar_v2/QRadar_v2.py

```diff
@@ -0,0 +1,1961 @@
+        }
+    obj = filter_dict_null(obj)
+    if headers:
+        if isinstance(headers, str):
+            headers = headers.split(",")
+        if isinstance(obj, dict):
+            headers = list(set(headers).intersection(set(obj.keys())))
+    ec = {context_key: obj} if context_key else obj
+    return {
+        "Type": entryTypes["note"],
+        "Contents": contents,
+        "ContentsFormat": formats["json"],
+        "ReadableContentsFormat": formats["markdown"],
+        "HumanReadable": human_readable
+        if human_readable
+        else tableToMarkdown(title, obj, headers).replace("\t", " "),
+        "EntryContext": ec,
+    }
+
+
+def epoch_to_iso(ms_passed_since_epoch):
+    """
+    Converts epoch (miliseconds) to ISO string
+    """
+    if ms_passed_since_epoch >= 0:
+        return datetime.utcfromtimestamp(ms_passed_since_epoch / 1000.0).strftime(
+            "%Y-%m-%dT%H:%M:%S.%fZ"
+        )
+    return ms_passed_since_epoch
+
+
+def print_debug_msg(msg, lock: Lock = None):
+    """
+    Prints a debug message with QRadarMsg prefix, while handling lock.acquire (if available)
+    """
+    err_msg = f"QRadarMsg - {msg}"
+    if lock:
+        if lock.acquire(timeout=LOCK_WAIT_TIME):
+            demisto.debug(err_msg)
+            lock.release()
+    else:
+        demisto.debug(err_msg)
+
+
+def filter_dict_null(d):
+    """
+    Filters recursively null values from dictionary
+    """
+    if isinstance(d, dict):
+        return dict(
+            (k, filter_dict_null(v))
+            for k, v in list(d.items())
+            if filter_dict_null(v) is not None
+        )
+    elif isinstance(d, list):
+        if len(d) > 0:
+            return list(map(filter_dict_null, d))
+        return None
+    return d
+
+
+def filter_dict_non_intersection_key_to_value(d1, d2):
+    """
+    Filters recursively from dictionary (d1) all keys that do not appear in d2
+    """
+    if isinstance(d1, list):
+        return [filter_dict_non_intersection_key_to_value(x, d2) for x in d1]
+    elif isinstance(d1, dict) and isinstance(d2, dict):
+        d2values = list(d2.values())
+        return dict((k, v) for k, v in list(d1.items()) if k in d2values)
+    return d1
+
+
+def replace_keys(src, trans_map):
+    """
+    Change the keys of a dictionary according to a conversion map
+    trans_map - { 'OldKey': 'NewKey', ...}
+    """
+
+    def replace(key, trans_map_):
+        if key in trans_map_:
+            return trans_map_[key]
+        return key
+
+    if trans_map:
+        if isinstance(src, list):
+            return [replace_keys(x, trans_map) for x in src]
+        else:
+            src = {replace(k, trans_map): v for k, v in src.items()}
+    return src
+
+
+def dict_values_to_comma_separated_string(dic):
+    """
+    Transforms flat dictionary to comma separated values
+    """
+    return ",".join(str(v) for v in dic.values())
+
+
+""" Command functions """
+
+
+def test_module(client: QRadarClient):
```

Comment:
I can add a check for `offenses` fetch if `longRunning` is true - this is the only way to trigger fetch-incidents in this integration. I will also add a check if the integration can create a fetch search correctly (this will ensure that the supplied params can fetch successfully).